### DISCUSSION TITLE
How to obtain material properties from three time steps ago?

### DISCUSSION THREAD

**Discussion by tttsss-01** [idaholab/moose#24884] [Q&A General] [answered]
`getMaterialPropertyOld` and `getMaterialPropertyOlder` can retrieve material properties from the previous one and two time steps, respectively. How can I obtain material properties from three time steps ago or even earlier?

**Comment by GiudGiud** [comment]
Hello

I m afraid we dont have that one!
You could hack it though.
Use a material property which's sole job is to store the old property of another material property.
Then older for that one gives you the 3x old of the one you care about.

@lindsayad for awareness

Guillaume

**Comment by loganharbour** [comment, marked as answer]
We can actually do this with #24673 (merged) and a small change to what the max state allowed is (currently "older"))

What is the use case here? Why do you need such old properties?
